src/udf/wit/udf.wit

```diff
@@ -0,0 +1,49 @@
+package risingwave:udf // Maybe we can use a different package name? udf as a world name is enough.
+
+world udf {
+  use types.{schema, record-batch, eval-errno}
+
+  // TODO: is schema needed? since record-batch already contains schema.
+  export input-schema: func() -> schema
+  export output-schema: func() -> schema
+
+  // export init: func(inputs: list<scalar>) -> result<_, init-errno>
+  export eval: func(batch: record-batch) -> result<record-batch, eval-errno>
+}
+
+interface types {
+  // in Arrow IPC Streaming Format
+  type record-batch = list<u8>
+
+  enum data-type {
+    dt-i16,
+    dt-i32,
+    dt-i64,
+    dt-bool,
+    dt-string,
+  }
+
+  type schema = list<field>
+
+  record field {
+    name: string,
+    data-type: data-type,
+  }
+
+  enum init-errno {
+    invalid-params,
+  }
+
+  enum eval-errno {
+    numeric-overflow,
+    division-by-zero,
+  }
+
+  union scalar {
+    s16,
+    s32,
+    s64,
+    bool,
+    string,
+  }
```

Comment:
FYI The `union` block was removed in newer version. 😇
https://github.com/WebAssembly/component-model/commit/8f0a9175d5d3982c14aab581fc56a73552c9f74c

Reply:
btw fyi, the wit interface `udf.wit` is also very arbitrary, not carefully designed 🤪

Reply:
Just removed since it's not being used now.